charts/kong/templates/admission-webhook.yaml

```diff
@@ -1,7 +1,20 @@
-{{- if .Values.ingressController.admissionWebhook.enabled }}
-{{- $cn := printf "%s.%s.svc" ( include "kong.service.validationWebhook" . ) ( include "kong.namespace" . ) }}
+{{- if .Values.ingressController.admissionWebhook.enabled -}}
+{{- $cn := printf "%s.%s.svc" ( include "kong.service.validationWebhook" . ) ( include "kong.namespace" . ) -}}
 {{- $ca := genCA "kong-admission-ca" 3650 -}}
 {{- $cert := genSignedCert $cn nil nil 3650 $ca -}}
+{{- $certCert := $cert.Cert -}}
+{{- $certKey := $cert.Key -}}
+{{- $caCert := $ca.Cert -}}
+{{- $caKey := $ca.Key -}}
```

Comment:
Should this be in the `else` branch of the `if $certSecret`?

I don't see why we should generate certs if the secret is already present.

Maybe this is the reason for 
> why the install/upgrade checksum thing happens

?

Reply:
No particular reason. I didn't move it from its original location and it provides something to initialize the certificate variables to.

This shouldn't affect the checksum--as best I know, that's generated from the text output of the template only, so the variable initialization at the top shouldn't matter, as it doesn't output text and prunes all whitespace before `kind: ValidatingWebhookConfiguration`.

The result looks identical in the final output, though that makes some changes to the individual templates' output (it sorts resources at least--not sure of all the changes it makes). There's presumably something that's either pruned by that pass or something that's different when using it as a named template (the SHA input) versus as a rendered file (the content we pass to K8S and see in Helm debug output). Unfortunately Helm doesn't provide an easy way of seeing the former.